contracts/LoanCore.sol

```diff
@@ -743,12 +800,20 @@ contract LoanCore is
         if (protocolFee > 0) _feesWithdrawable[address(this)] += protocolFee;
         if (affiliateFee > 0) _feesWithdrawable[affiliate] += affiliateFee;
 
-        // State changes and cleanup
-        loans[loanId].state = LoanLibrary.LoanState.Repaid;
-        loans[loanId].interestAmountPaid = _amountFromPayer - data.balance;
-        loans[loanId].balance = 0;
+        // state changes
+        loanRepaid = false;
```

Comment:
Can save an assignment here by just having `loanRepaid = (loans[loanId].state == LoanLibrary.LoanState.Repaid)` after the `if` block.

Or, don't return `LoanRepaid` and just have callers of _handleRepay read the `state` from `LoanData` (can ignore this if we remove that enum altogether)

Reply:
Yes good point, I removed the local var and changed to the calling functions looking at the LoanState enum